### DISCUSSION TITLE
What are the key differences between redirect, router.push, and window.location.assign in the context of Next.js applications?

### DISCUSSION THREAD

**Discussion by kemaltf** [vercel/next.js#59689] [Help]
### Summary

When developing in Next.js, how do the use cases for redirect, router.push, and window.location.assign differ when it comes to navigating between pages? since all that function is used for directing us to another URL i think.

Thanks for your help!

**Comment by nirvikpurkait** [comment]
Basically `redirect` is used to navigate between `url`'s on server or server action, `router.push()` is used in client side navigation, and ` window.location.assign` can be used in all over JS, even you can access them on browser console.
